packages/vega4-extension/src/index.ts

```diff
@@ -153,3 +157,46 @@ const extension: IRenderMime.IExtension = {
 };
 
 export default extension;
+
+/**
+ * A namespace for private module data.
+ */
+namespace Private {
+  /**
+   * A cached reference to the vega library.
+   */
+  export let vega: typeof VegaModuleType;
+
+  /**
+   * A Promise for the initial load of vega.
+   */
+  export let vegaReady: Promise<typeof VegaModuleType>;
+
+  /**
+   * Lazy-load and cache the vega-embed library
+   */
+  export function ensureVega(): Promise<typeof VegaModuleType> {
+    if (vegaReady) {
+      return vegaReady;
+    }
+
+    vegaReady = new Promise((resolve, reject) => {
+      require.ensure(
```

Comment:
Hooray, the `///` directive makes this work... most of it is still untyped though because of how crazy this syntax is...